### PR TITLE
Improve code related to CG

### PR DESCRIPTION
**Pull request by paraynaud**
Try to solve #21

### CODE REVIEW

**Comment by paraynaud** [comment]
@dpo can you review this one?

File: src/krylov.jl [hunk 1]
```diff
@@ -92,7 +114,12 @@ function axpby!(
   ::Val{true},
   ::Val{true},
 ) where {T <: Number, Y1 <: Number, Y2 <: Number}
-  y .= x .* s .+ y .* t
```
Comment (dpo) [inline src/krylov.jl:95]: I don't understand how the two `Val` arguments work. Sometimes they're `true` (if `x.simulate_vector` is `true`), and you call `build!()`, but not always.

**Comment by paraynaud** [comment]
In fact there is no allocations in `solve!(::CGSolver)`.
I made a mistake, I performed 
```julia
@benchmark Krylov.solve!(::CgSolver, ::PartitionedLinearOperator, -pv_gradient)
```
the allocations came from `-pv_gradient` which allocates a new `PartitionedVector`.
```julia
pvb = -pv_gradient
@benchmark Krylov.solve!(solver, lo_epm, pvb)
BenchmarkTools.Trial: 10000 samples with 1 evaluation.
 Range (min … max):  25.700 μs … 228.600 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     26.300 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   29.780 μs ±  13.866 μs  ┊ GC (mean ± σ):  0.00% ± 0.00%
  █▃▂▂▁▁       ▁  ▁▁                                           ▁
  ██████▇▆▆▅▅▅▆██▅██▆▆▅▅▄▄▅▄▄▆▆▇█▇▇▆▇▅▄▅▅▆▄▅▅▅▄▁▄▅▄▁▃▄▁▅▄▄▅▆▆▅ █
  25.7 μs       Histogram: log(frequency) by time       103 μs <
 Memory estimate: 96 bytes, allocs estimate: 4.
```
for a small example.
I changed the partitioned structure of the partitioned linear problem and there was no additional allocations.

**Comment by dpo** [comment]
ok so is this good to go?

**Comment by paraynaud** [comment]
> ok so is this good to go?

Yes

**Review by dpo — APPROVED** [review]

**Pull request merged**